### PR TITLE
Add more default metrics for Python topology

### PR DESCRIPTION
**Pull request by objmagic**
This patch adds more default metrics for Python topology, namely:

```
'__gateway-sent-metrics-size'
'__gateway-sent-metrics-packets-count'
'__gateway-sent-metrics-count'
'__gateway-sent-exceptions-count'
'__gateway-in-stream-queue-size'
'__gateway-out-stream-queue-size'
'__gateway-in-stream-queue-expected-capacity'
'__gateway-out-stream-queue-expected-capacity'
'__gateway-in-queue-full-count'
```

This PR also adds a half-acking topology example.

related to #1264

### CODE REVIEW

**Comment by kramasamy** [comment]
@taishi8117 - if you could take a look at this, it will be awesome.

**Comment by kramasamy** [comment]
it looks fine to me 👍

**Comment by kramasamy** [comment]
let us wait until @taishi8117 gives feedback.

File: heron/examples/src/python/half_acking_topology.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Refer to multi_stream_topology for defining a topology by subclassing Topology
+if __name__ == '__main__':
+  if len(sys.argv) != 2:
+    print "Topology's name is not specified"
+    sys.exit(1)
+
+  builder = TopologyBuilder(name=sys.argv[1])
+
+  word_spout = builder.add_spout("word_spout", WordSpout, par=2)
+  count_bolt = builder.add_bolt("half_ack_bolt", HalfAckBolt, par=2,
```
Comment (taishi8117) [inline heron/examples/src/python/half_acking_topology.py]: maybe change the variable name to `half_ack_bolt`?

File: heron/instance/src/python/instance/st_heron_instance.py [hunk 2]
```diff
@@ -250,7 +252,7 @@ def main():
   max_log_bytes = system_config[constants.HERON_LOGGING_MAXIMUM_SIZE_MB] * constants.MB
 
   log_file = os.path.join(log_dir, instance_id + ".log.0")
-  log.init_rotating_logger(level=logging.INFO, logfile=log_file,
+  log.init_rotating_logger(level=logging.DEBUG, logfile=log_file,
```
Comment (taishi8117) [inline heron/instance/src/python/instance/st_heron_instance.py]: We should change the default logging level back to `INFO`

**Comment by taishi8117** [comment]
@kramasamy @objmagic I made some comments, but other than that, this PR looks fine to me.

**Comment by objmagic** [comment]
@taishi8117 I have addressed your comments in the latest commit.

**Comment by objmagic** [comment]
seems integration test is very unstable. restarted CI...

**Comment by kramasamy** [comment]
Can you identify which integration test is unstable?

**Comment by objmagic** [comment]
seems to me each one of them failed before (retried > 25 times)

**Comment by objmagic** [comment]
restarted CI and now it's fine.

**Pull request merged**